### ISSUE TITLE
Mamba3-SISO exploding gradients on B200

### ISSUE THREAD

**Issue by NadavSc** [state-spaces/mamba#938]
Hi!
I have encountered some gradients exploding while training on B200.
As you can see, the training starts the same but after a while it explodes, contrary to the L40s training which remains stable.
I trained the model on the-pile with 1 GPU and ~60k tokens per step. AdamW optimizer with weight_decay of 0.1.

<img width="368" height="272" alt="Image" src="https://github.com/user-attachments/assets/4ab3079d-5dcd-403d-8fd8-f353b146d611" />

<img width="370" height="264" alt="Image" src="https://github.com/user-attachments/assets/7a51b61d-0edb-44b1-b9da-9252715097ad" />

**Comment by OMGYFX** [comment]
Buddy js use Gradient Clipping: weight_decay=0.1

**Comment by NadavSc** [comment]
> Buddy js use Gradient Clipping: weight_decay=0.1

I was already using gradient clipping and weight decay of 0.1.
Reducing chunk size to 32 instead of 64 fixed that problem.

**Comment by aakashlahoti** [comment]
HI Nadav! This turned out to be an issue with Triton compiler on Blackwell GPUs with num_stages > 1. Please check: https://github.com/state-spaces/mamba/pull/997/changes for a fix.
